geoportal/tests/functional/__init__.py

```diff
@@ -76,27 +79,27 @@ def cleanup_db() -> None:
     )
     from c2cgeoportal_commons.models.static import OAuth2Client, Shorturl, User
 
-    try:
-        transaction.commit()
-    except sqlalchemy.exc.InvalidRequestError:
-        models.DBSession.rollback()
-
-    for ra in models.DBSession.query(RestrictionArea).all():
-        ra.roles = []
-        models.DBSession.delete(ra)
-    for ti in models.DBSession.query(TreeItem).all():
-        models.DBSession.delete(ti)
-    models.DBSession.query(OGCServer).delete()
-    models.DBSession.query(Interface).delete()
-    for r in models.DBSession.query(Role).all():
-        r.functionalities = []
-        models.DBSession.delete(r)
-    models.DBSession.query(User).delete()
-    models.DBSession.query(Functionality).delete()
-    models.DBSession.query(FullTextSearch).delete()
-    models.DBSession.query(Shorturl).delete()
-    models.DBSession.query(OAuth2Client).delete()
-    transaction.commit()
+    with models.DBSession() as session:
```

Comment:
IMHO we should completely avoid using the global DBSession object and use a fixture instead, but I may miss something.